libraries/chain/webassembly/runtimes/eos-vm-oc/executor.cpp

```diff
@@ -262,7 +263,17 @@ void executor::execute(const code_descriptor& code, memory& mem, apply_context&
                   start_func();
                }
             }, code.start);
-            apply_func(context.get_receiver().to_uint64_t(), context.get_action().account.to_uint64_t(), context.get_action().name.to_uint64_t());
+
+            if (context.is_action()) {
+               void(*apply_func)(uint64_t, uint64_t, uint64_t) = (void(*)(uint64_t, uint64_t, uint64_t))(cb->running_code_base + code.apply_offset);
+               apply_func(context.get_receiver().to_uint64_t(), context.get_action().account.to_uint64_t(), context.get_action().name.to_uint64_t());
+            } else if (code.call_offset) {
+               ilog("here");
+               int64_t(*call_func)(uint64_t, uint64_t, uint32_t) = (int64_t(*)(uint64_t, uint64_t, uint32_t))(cb->running_code_base + *code.call_offset);
+               call_func(context.get_sender().to_uint64_t(), context.get_receiver().to_uint64_t(), static_cast<uint32_t>(static_cast<eosio::chain::sync_call_context&>(context).data.size()));
+            } else {
+               status = eosio::chain::execution_status::receiver_not_support_sync_call;
```

Comment:
How to check `sync_call` entry point signature?

Reply:
imo we ought to not need to. The validity of the sync_call export doesn't need to be tracked and checked by each individual wasm runtime in a bespoke way -- we ought to be able to that in a single common location.

Unfortunately it seems we cannot use the instantiation cache as is, since OC can bypass that.

We could use the code index but perhaps that is considered improper. (that said, the code index already tracks non-protocol aspects of code like a reference count!)

I don't have other ideas at the moment.

Reply:
Created an issue https://github.com/AntelopeIO/spring/issues/893